### PR TITLE
protected pages and passed cookie when creating img

### PR DESCRIPTION
**Pull request by Gannon44**
* Changed isAuthenticated function in auth.js to call backend.
* Added backend route for authenticating user
* Added page protection function on most pages (not register, login, or leaderboard)
* Create page now passes cookie on submit

Each page now routes as follows if jwt is authenticated:
* (page -> on_success | on_fail)
* portfolio -> portfolio | login
* create -> create | login
* vote -> vote | login
* / -> create | /           (this one only lets you view the / page if you aren't signed in)

Please try running the code!! Everything ran successfully on my local machine, but I changed a lot of files.

### CODE REVIEW

File: Frontend/src/pages/create.js [hunk 1]
```diff
@@ -114,3 +121,24 @@ export default function Create() {
...
+    // If the user is not authenticated, redirect them to the login page
+    return {
+      redirect: {
+        destination: "/login",
+        permanent: false,
+      },
+    };
+  }
+
+  // If the user is authenticated, render the Portfolio page
```
Comment (nnayk) [inline Frontend/src/pages/create.js:140]: Is this an old comment? Confused about the "render Portfolio page" part

**Review by nnayk — APPROVED** [review]
Nice job on protecting the routes + fixing the generate_image endpoint issue. I added a note about one comment that didn't make sense to me. Also, I know we discussed the repeated getServerSideProps() code. I'm ok with this for now but might be something worth looking into addressing if time permits. Overall great work.

File: Frontend/src/pages/create.js (hunk 1 above)
Comment (Gannon44) [reply Frontend/src/pages/create.js:140]: its due to me copy and pasting from the portfolio page... should say "Create page"

**Review by caseyavila — APPROVED** [review]
Works well on my side, good work!

**Pull request merged**